vendor/milesight-iot/ws301.js

```diff
@@ -49,7 +49,7 @@ function normalizeUplink(input) {
     return {
         data: {
             action: {
-                contactState: input.data.door.toUpperCase()
+                contactState: input.data.door ? "OPEN":"CLOSED"
```

Comment:
So is it a boolean? Before you assumed it was a string?

Reply:
So in this case `input.data.door` is either `close` or `open`. Not `CLOSE` or `OPEN` and not a boolean.

To convert `close` or `open` to `CLOSED` or `OPEN` respectively, you need to `contactState: input.data.door === "close" ? "CLOSED" ? input.data.door === "open ? "OPEN" : undefined`

Reply:
I finally see how it should work now, apologies.